src/plugin/memory-webview-main.ts

```diff
@@ -84,9 +87,52 @@ export class MemoryWebview {
                 }
             })
         );
+
+        vscode.window.registerCustomEditorProvider(manifest.EDITOR_NAME, this);
     };
 
-    public async show(initialMemory?: Partial<DebugProtocol.ReadMemoryArguments>): Promise<void> {
+    public async saveCustomDocument(_document: vscode.CustomDocument, _cancellation: vscode.CancellationToken): Promise<void> {}
+    public async saveCustomDocumentAs(_document: vscode.CustomDocument, _destination: vscode.Uri, _cancellation: vscode.CancellationToken): Promise<void> {}
+    public async revertCustomDocument(_document: vscode.CustomDocument, _cancellation: vscode.CancellationToken): Promise<void> {}
+    public async backupCustomDocument(
+        _document: vscode.CustomDocument,
+        _context: vscode.CustomDocumentBackupContext,
+        _cancellation: vscode.CancellationToken
+    ): Promise<vscode.CustomDocumentBackup> {
+        return {
+            id: '',
+            delete: () => {}
+        };
+    }
+
+    public openCustomDocument(uri: vscode.Uri, _openContext: vscode.CustomDocumentOpenContext, _token: vscode.CancellationToken): vscode.CustomDocument {
+        return {
+            uri,
+            dispose: () => {}
+        };
+    }
+
+    public async resolveCustomEditor(document: vscode.CustomDocument, webviewPanel: vscode.WebviewPanel, _token: vscode.CancellationToken): Promise<void> {
+        /*
+            memoryReference = debugprotocol.variable.memoryReference
+            displayName = 'memory'
+            DEBUG_MEMORY_SCHEME = 'vscode-debug-memory'
+            sessionId = <debug session ID>
+            range = undefined
+
+            document.uri is:
+
+            scheme: DEBUG_MEMORY_SCHEME,
+            authority: sessionId,
+            path: '/' + encodeURIComponent(memoryReference) + `/${encodeURIComponent(displayName)}.bin`,
+            query: range ? `?range=${range.fromOffset}:${range.toOffset}` : undefined,
```

Comment:
Perhaps worth commenting on any future work on this in the VSCode repository that using elements of the path here is probably not ideal, since they're unlabled and therefore inherently hard to document. Likely all DAP-relevant data should be encoded as query components.

Reply:
Yeah, its not nice!